share/getters/store.go

```diff
@@ -110,44 +114,5 @@ func (sg *StoreGetter) GetSharesByNamespace(
 		utils.SetStatusAndEnd(span, err)
 	}()
 
-	if err = namespace.ValidateForData(); err != nil {
-		return nil, err
-	}
-
-	bs, err := sg.store.CARBlockstore(ctx, root.Hash())
-	if errors.Is(err, eds.ErrNotFound) {
-		// convert error to satisfy getter interface contract
-		err = share.ErrNotFound
-	}
-	if err != nil {
-		return nil, fmt.Errorf("getter/store: failed to retrieve blockstore: %w", err)
-	}
-	defer func() {
-		if err := bs.Close(); err != nil {
-			log.Warnw("closing blockstore", "err", err)
-		}
-	}()
-
-	// wrap the read-only CAR blockstore in a getter
-	blockGetter := eds.NewBlockGetter(bs)
-	shares, err = collectSharesByNamespace(ctx, blockGetter, root, namespace)
-	if errors.Is(err, ipld.ErrNodeNotFound) {
-		// IPLD node not found after the index pointed to this shard and the CAR
-		// blockstore has been opened successfully is a strong indicator of
-		// corruption. We remove the block on bridges and fulls and return
-		// share.ErrNotFound to ensure the data is retrieved by the next getter.
-		// Note that this recovery is manual and will only be restored by an RPC
-		// call to SharesAvailable that fetches the same datahash that was
-		// removed.
-		err = sg.store.Remove(ctx, root.Hash())
-		if err != nil {
-			log.Errorf("getter/store: failed to remove CAR after detected corruption: %w", err)
-		}
-		err = share.ErrNotFound
-	}
-	if err != nil {
-		return nil, fmt.Errorf("getter/store: failed to retrieve shares by namespace: %w", err)
-	}
-
-	return shares, nil
+	return eds.RetrieveNamespaceFromStore(ctx, sg.store, header.DAH, namespace)
```

Comment:
Would it be possible to identfy that error happened in store getter and not in shrexnd server? I would suggest to wrap it here just in case, so later we don't have to worry to double check, that caller wraps it with identifiable text